x-pack/plugins/ingest_manager/public/plugin.ts

```diff
@@ -22,7 +22,12 @@ import { registerDatasource } from './applications/ingest_manager/sections/agent
 
 export { IngestManagerConfigType } from '../common/types';
 
-export type IngestManagerSetup = void;
+// We need to provide an object instead of void so that dependent plugins know when Ingest Manager
+// is disabled.
+export interface IngestManagerSetup {
```

Comment:
@elastic/ingest-management I made this change so we could inspect whether Ingest Manager was disabled or not. When it's disabled, the Ingest Manager setup dependency will be `undefined`. I'm happy to change this return value to anything you'd prefer.